Gordon360/Services/RecIM/SportService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Models.ViewModels.RecIM;
+using Gordon360.Models.CCT.Context;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Services.RecIM
+{
+    public class SportService : ISportService
+    {
+        private readonly CCTContext _context;
+
+        public SportService(CCTContext context)
+        {
+            _context = context;
+        }
+
+        public SportViewModel GetSportByID(int sportID)
+        {
+            return _context.Sport
+                    .FirstOrDefault(s => s.ID == sportID);
+        }
+
+        public IEnumerable<SportViewModel> GetSports()
+        {
+            return _context.Sport.Select(s => (SportViewModel)s).AsEnumerable();
+        }
+
+        public async Task<int> PostSport(SportUploadViewModel newSport)
+        {
+            var sport = new Sport
+            {
+                Name = newSport.Name,
+                Description = newSport.Description,
+                Rules = newSport.Rules,
+                Logo = newSport.Logo ?? "defaultSportPicture.jpg" //will be a good idea to make default pictures
+            };
+            await _context.Sport.AddAsync(sport);
+            await _context.SaveChangesAsync();
+            return sport.ID;
+        }
+
+        public async Task UpdateSport(SportViewModel updatedSport)
```

Comment:
Take in an `int sportID` here.

Reply:
Also, since you are patching here, perhaps it would be better to name this method `PatchSport(...)`?

Reply:
> Also, since you are patching here, perhaps it would be better to name this method `PatchSport(...)`?

There is no concept of patching in the service here, this is just updating the database row.

Just like we wouldn't name this `PutSport` if it was called by a PUT route or `PostSport` if it was called by a POST route.

Reply:
I was thinking of patch more as a general term than a HTTP term. As I mentioned above, our existing Update...() service methods should really be Replace...() service methods, but not an issue we will fix here. @amos-cha disregard the second comment in this thread as well.